### PR TITLE
fix(harbor):fix miss Issuer for registry cert

### LINKED ISSUE #124: Miss Issure for registry cert
Miss create `Issuer` for registry cert in Harbor CR namespace.

image

Need to use YAML to create `Issuer`.
```yaml
apiVersion: cert-manager.io/v1alpha2
kind: Issuer
metadata:
  name: selfsigned-issuer
  namespace: pg
spec:
  selfSigned: {}
```

### PR DESCRIPTION
**Pull request by Ghostbaby**
Add check function to make sure Issuer will be created.

Fixes #124

Signed-off-by: zhuhuijun <zhuhuijunzhj@gmail.com>

### CODE REVIEW

File: controllers/harbor/cert.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+			return err
+		}
+	}
+	return nil
+}
+
+// provisionIssuer will create issuer for registry
+func provisionIssuer(client k8s.Client, name, namespace string) error {
+	issuer := newIssuer(name, namespace)
+	err := client.Create(issuer)
```
Comment (steven-zou) [inline controllers/harbor/cert.go:39]:
```suggestion
	return client.Create(issuer)
```

File: controllers/harbor/cert.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+// CheckIssuer check issuer has exist, if not will create issuer
+func (harbor *HarborReconciler) CheckIssuer() error {
+	var issuer certv1.Issuer
+	namespacedName := harbor.getRegistryIssuerNamespacedName()
+
+	err := harbor.Get(namespacedName, &issuer)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return provisionIssuer(harbor.Client, namespacedName.Name, namespacedName.Namespace)
+		} else {
```
Comment (steven-zou) [inline controllers/harbor/cert.go:29]:
```suggestion
		
```

File: controllers/harbor/cert.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+			return err
+		}
+	}
+	return nil
```
Comment (steven-zou) [inline controllers/harbor/cert.go:33]:
```suggestion
	return err
```

Comment (steven-zou) [reply controllers/harbor/cert.go:33]: together with above comment

**Comment by wangcanfengxs** [comment]
LGTM

**Pull request merged**